tensorflow/python/ops/math_ops.py

```diff
@@ -2786,7 +2786,19 @@ def add_n(inputs, name=None):
   """Adds all input tensors element-wise.
 
   Converts `IndexedSlices` objects into dense tensors prior to adding.
-
+  
+  `tf.math.add_n` performs the same operation as `tf.math.accumulate_n` but it waits
+  for all of its inputs to be ready before beginning to sum, that consumes more 
+  memory as  inputs are not ready at different times, since minimum temporary storage 
```

Comment:
> that consumes more memory as  inputs are not ready at different times

I'm not sure this makes sense. How about "This can consume more memory when inputs are ready at different times, since the minimum temporary storage required is proportional to the input size rather than the output size."

Reply:
Basically, in tf.math.accumulate_n documentation anti  of this is given as the reason for it to be better than tf.math.add_n, you can check there.So I just manipulated that statement so that why tf.math.add_n is less preferrable than accumulate_n.

Reply:
Ok still I am adding what you just said.